sdk/src/main/java/io/opentelemetry/sdk/trace/config/TraceConfig.java

```diff
@@ -48,25 +89,17 @@
   private static final int DEFAULT_SPAN_MAX_NUM_ATTRIBUTES_PER_LINK = 32;
 
   /**
-   * Returns the default {@code TraceConfig}.
+   * Returns a new {@link TraceConfig} reading the configuration values from the environment and
+   * from system properties. System properties override values defined in the environment. If a
+   * configuration value is missing, it uses the default value.
    *
    * @return the default {@code TraceConfig}.
    * @since 0.1.0
    */
   public static TraceConfig getDefault() {
-    return DEFAULT;
+    return newBuilder().readEnvironmentVariables().readSystemProperties().build();
```

Comment:
Usually `getDefault()` methods return a a static, read-only value. I wonder if we should have the same here, or else have this method use a different name.

Reply:
Also: do we really want to use env values as default here? I know that for `Resources` we do environment-variables fetching by default, and I'm wondering if we should do it here as well.

Reply:
Agreed. A method called 'getDefault' should be returning a trace config with the default settings, not returning one that has been configured by the user.

Reply:
Yes. Agreed. `getDefault` is less intuitive. If `getDefault` returns trace config with the default settings, anybody who requires the config to be read from either env vars or sys props should again use `updateActiveTraceConfig` as default TraceConfig instance is assigned here. https://github.com/open-telemetry/opentelemetry-java/blob/master/sdk/src/main/java/io/opentelemetry/sdk/trace/TracerSharedState.java#L35

My proposal is to keep the `getDefault` similar to how we do for `Resources` where we fetch env vars by default or if `getDefault` is less intuitive, we can rename it as simply `getInstance`. what do you think?

Reply:
I would argue the other way. I think we should change the method on Resources to make it clear it's being loaded from the environment, rather than perpetuate the confusing naming to more places.

Reply:
+1 on changing `Resources` to make it clear we are loading stuff from the environment.

Reply:
Thanks for the reply. There are 2 things to work on.

- Shall we have one more method `getInstance` to fetch settings from env vars/sys props to use it across resources?
- Is `getDefault` still required? shall we remove/deprecate this? Because, `getInstance` will provide trace config with default settings only in the absence of env vars and sys props.

Please let me know how to proceed further.

Reply:
For the moment, revert the implementation of `getDefault` to what it was before. If we want to provide additional methods that bypass the builder, we can make that decision separately, and apply it across the board to all the configurable items in the SDK.

Reply:
Done.